### DISCUSSION TITLE
How to deploy front-end separately from back ?

### DISCUSSION THREAD

**Discussion by rramillien** [tchiotludo/akhq#1255] [General]
Hello,

We would like to deploy AKHQ in our enterprise, but in 2 differents areas.
Not really this, but just for example, you can think of deploying front in DMZ and backend on back area.

From what I see, there is only one fat JAR to download which contains both front and back.
Is there a prepackaged front and back applications available anywhere or any documentation on building them separately ?

Best regards,

Régis

**Comment by tchiotludo** [comment]
No there is not for now, and I will never do that since it don't have any sense for the project.
The thing I could suggest you will be to extract the fat jar to have the ui and deployed it like you want. 
The backend, behind a reverse proxy or using micronaut configuration, can be disabled I think overloading [this](https://github.com/tchiotludo/akhq/blob/b732a16675a250fea1c91e2f3311798350130f89/src/main/resources/application.yml#L55-L61) configuration

**Reply by rramillien** [reply]
Hello Ludo,

Thanks a lot for your quick response.
But I do not understand when you say: "it don't have any sense for the project" ?
In my understanding, kafka is almost always installed in the network back area. These areas are more secured than front ones and filter most of the requests than the ones in DMZ. Applications in a back areas are not directly exposed to internet but have intermediate middlewares (firewalls, API managers, etc).
So, when we a user wants to connect to AKHQ, it should connect "directly" to the front, and we should not have to open ports for UI or configure an API manager to access it. Only ports for the API (API manager) should be opened to internet.
Is there something special about AKHQ that requires it to be in a back area ?

Best regards,

Régis

**Reply by tchiotludo** [reply]
The backend (AKHQ api need a direct connection) need a direct connection on the cluster. 
The front only need a direct connection to backend (api) 

> have any sense for the project
mean that for the project himself, it not really a use case to be powered by himself. You could have some security constraint for your company and you could deal it easily with reverse proxy or api manager. 

In general, most of the implementation is simply based on a reverse proxy, akhq is fully on the backend side (akhq api + akhq frontend) and the front (exposed to internet) is a simple reverse proxy to akhq backend.

**Reply by rramillien** [reply]
Hello Ludo,

Ok, thank you for your response.
